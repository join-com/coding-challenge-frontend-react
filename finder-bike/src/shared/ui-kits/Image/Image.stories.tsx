import { storiesOf } from '@storybook/react';
import React from 'react';
import Image from "./Image"

storiesOf("UI-Kit/Image", module)
  .add("with src", () => (
    <Image src="https://seeclickfix.com/files/issue_images/0137/6393/img-image_480745259766912201753441.jpg" />
  ))