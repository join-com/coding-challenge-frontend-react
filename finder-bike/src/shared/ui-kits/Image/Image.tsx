import React from 'react';
import styled from 'styled-components';
import noImage from '../../../assets/images/no-image.png';

const StyledImage = styled.img`
  width: 100%;
`;

function Image({ src, ...props }) {
  return <StyledImage src={src ? src : noImage} {...props} />;
}
export default Image;
