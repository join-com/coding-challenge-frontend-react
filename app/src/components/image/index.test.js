import React from 'react'
import Image from './index'
import renderer from 'react-test-renderer'
import { shallow, render } from 'enzyme'

test('Should render image component with image not available', () => {
  const component = renderer.create(<Image />)
  let tree = component.toJSON()
  expect(tree).not.toBeNull()
  expect(component).toMatchSnapshot()
})

test('Should render image component with image src', () => {
  const component = render(<Image src={'/path/to/image'} />)
  expect(component).toMatchSnapshot()
})

test('Should render image component with image src and style', () => {
  const component = render(<Image src={'/path/to/image'} style={{ maxWidth: '50px' }} />)
  expect(component).toMatchSnapshot()
})
