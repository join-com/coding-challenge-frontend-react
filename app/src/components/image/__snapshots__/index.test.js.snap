// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Should render image component with image not available 1`] = `
<img
  alt="not available"
  className="img-thumbnail img-fluid"
  src="not_available.jpg"
/>
`;

exports[`Should render image component with image src 1`] = `
<img
  alt="not available"
  class="img-thumbnail img-fluid"
  src="/path/to/image"
/>
`;

exports[`Should render image component with image src and style 1`] = `
<img
  alt="not available"
  class="img-thumbnail img-fluid"
  src="/path/to/image"
  style="max-width:50px"
/>
`;
